File: R/1402-reducing-dishes/main.go
package main

import (
	"fmt"
	"sort"
)

func maxSatisfaction(satisfaction []int) int {
	sort.Ints(satisfaction)
	sum := 0
	accum := 0
	max := 0
	for i := len(satisfaction) - 1; i >= 0; i-- {
		accum += sum + satisfaction[i]
		sum += satisfaction[i]
		if accum > max {
			max = accum
		}
	}
	return max
}

func main() {
	fmt.Println()
}

File: R/1402-reducing-dishes/main.rs
struct Solution {}

impl Solution {
  pub fn max_satisfaction(satisfaction: Vec<i32>) -> i32 {
    let mut satisfaction = satisfaction;
    satisfaction.sort();
    let mut max: i32 = 0;
    let mut times: i32 = 1;
    let mut sum: i32 = 0;
    let mut aggr: i32 = 0;
    satisfaction.iter().rev().for_each(|x| {
      sum += x;
      aggr += sum;
      max = max.max(aggr);
    });
    max
  }
}
